internal/characters/zhongli/stele.go

```diff
@@ -113,23 +119,37 @@ func (c *char) resonance(src int) func() {
 
 			// queue stele attack
 			steleAttackPos := combat.CalcOffsetPoint(stelePos, boxOffset, steleDir)
-			c.Core.QueueAttackWithSnap(ai, snap, combat.NewBoxHitOnTarget(steleAttackPos, nil, boxSize, boxSize), 0)
+			c.Core.QueueAttackWithSnap(ai, snap, combat.NewBoxHitOnTarget(steleAttackPos, nil, boxSize, boxSize), 0, particleCB)
 
 			// queue resonance attacks
 			for _, con := range resonanceConstructs {
 				resonanceAttackPos := combat.CalcOffsetPoint(con.Pos(), boxOffset, con.Direction())
-				c.Core.QueueAttackWithSnap(ai, snap, combat.NewBoxHitOnTarget(resonanceAttackPos, nil, boxSize, boxSize), 0)
+				c.Core.QueueAttackWithSnap(ai, snap, combat.NewBoxHitOnTarget(resonanceAttackPos, nil, boxSize, boxSize), 0, particleCB)
 			}
+		}
+		c.Core.Tasks.Add(c.resonance(src), 120)
+	}
+}
 
-			// particle check
-			if c.energyICD < c.Core.F && !orb && c.Core.Rand.Float64() < .5 {
-				orb = true
-			}
+func (c *char) steleEnergyCB() combat.AttackCBFunc {
+	done := false
+	return func(a combat.AttackCB) {
+		if a.Target.Type() != combat.TargettableEnemy {
+			return
 		}
-		if orb {
-			c.energyICD = c.Core.F + 90
-			c.Core.QueueParticle("zhongli", 1, attributes.Geo, 20+c.ParticleDelay)
+		// can only proc once per skill cast / stele tick
```

Comment:
is done redundant since status will be added after first call (causing it to only proc once)?

Reply:
No, it's not because otherwise the other attacks in the stele tick will be able to take a shot at proccing that 50% chance.

Reply:
Nevermind.